viadot/flows/genesys_to_adls.py

```diff
@@ -1,30 +1,29 @@
-from typing import Any, Dict, List, Literal
+import os
+from typing import Any, Dict, List, Any
 
 import pandas as pd
 from prefect import Flow, task
 
-from viadot.tasks import AzureDataLakeUpload
-from viadot.tasks.genesys import GenesysToCSV, GenesysToDF
+from viadot.tasks.genesys import GenesysToCSV
 from viadot.task_utils import (
     add_ingestion_metadata_task,
-    df_to_csv,
-    df_to_parquet,
     adls_bulk_upload,
 )
 
 
-@task
-def add_timestamp(files_names: List = None, sep: str = None) -> None:
+@task(timeout=3600)
+def add_timestamp(files_names: List = None, path: str = "", sep: str = "\t") -> None:
     """Add new column _viadot_downloaded_at_utc into every genesys file.
 
     Args:
-        files_names (List, optional): All file names of downloaded files. Defaults to None.
+        files_names (List, optional): All file names of downloaded files. Defaults to "\t".
+        path (str, optional): Relative path to the file. Defaults to empty string.
         sep (str, optional): Separator in csv file. Defaults to None.
     """
     for file in files_names:
-        df = pd.read_csv(file, sep=sep)
+        df = pd.read_csv(os.path.join(path, file), sep=sep)
         df_updated = add_ingestion_metadata_task.run(df)
-        df_updated.to_csv(file, index=False, sep=sep)
+        df_updated.to_csv(os.path.join(path, file), index=False, sep=sep)
 
 
 class GenesysToADLS(Flow):
```

Comment:
Is there a way to generate dtypes (add a task for it)? You are currently uploading files to ADLS. In the next step - `toSQL` you have to specify dtypes. Maybe if you add `generating dtypes` it will be easier in the next step. Does that make sense or not so much?

Reply:
resolved ;)